Dockerfile

```diff
@@ -49,18 +48,19 @@ RUN \
 # copy sources
 COPY . /home/jhipster/generator-jhipster
 
+# When https://github.com/docker/docker/pull/28499 will be merged and released,
+# remove the following line and add "--chown jhipster:jhipster" to the previous one
 RUN \
   # fix jhipster user permissions
-  chown -R jhipster:jhipster \
-    /home/jhipster \
-    /usr/lib/node_modules && \
+  chown -R jhipster:jhipster /home/jhipster/generator-jhipster
 
+RUN \
   # install jhipster
-  rm -Rf /home/jhipster/generator-jhipster/node_modules \
+  rm -rf /home/jhipster/generator-jhipster/node_modules \
     /home/jhipster/generator-jhipster/yarn.lock \
     /home/jhipster/generator-jhipster/yarn-error.log && \
   su -c "cd /home/jhipster/generator-jhipster && yarn install" jhipster && \
-  su -c "yarn global add file:/home/jhipster/generator-jhipster" jhipster && \
+  yarn global add file:/home/jhipster/generator-jhipster && \
```

Comment:
It will probably work, but as mentionned, it will add another layer, so it will increase the size of the final Docker image.

You can look this commit: https://github.com/jhipster/generator-jhipster/commit/3940c9ef6664d00b2e11892d531c94551667024c#diff-3254677a7917c6c01f55212f86c57fbf
and the commit message:
> Optimize number of layers (https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices).

It is very interesting and we try to follow these best practices